pallets/omnipool/src/lib.rs

```diff
@@ -566,6 +443,13 @@ pub mod pallet {
 
 			ensure!(initial_price > FixedU128::zero(), Error::<T>::InvalidInitialAssetPrice);
 
+			// ensure collection is created, we can simply ignore the error if it was already created.
+			let _ = T::NFTHandler::create_collection(
+				&T::NFTCollectionId::get(),
+				&Self::protocol_account(),
+				&Self::protocol_account(),
+			);
+
```

Comment:
@mrq1911 i dont really like this here. 

I would remove it and let the creation be as prerequisite.